### ISSUE TITLE
SAML2 XML Configuration Digest Error

### ISSUE THREAD

**Issue by robertdeboer** [SocialiteProviders/Providers#860]
Hello all,

Thank you for this wonderful library. I have come across what I think is a bug when digesting a valid SAML 2 XML configuration generated by a Identity Provider for use in a service provider application.

Back story:
This is in used on a Laravel 8 application, using Laravel Socialite with Socialite Providers SAML2
```
"laravel/framework": "^8.12",
"laravel/socialite": "^5.5",
"socialiteproviders/saml2": "dev-master"
```
SAML2 is on dev-master simply so that I could utilize it's latest updates, which is switching over to litesaml/lightsaml from lightsaml/lightsaml. I think this issue is present in either master or a tag branch release.

This application is a simple Service Provider, with the Identity Provider being managed by a third party enterprise system. They provided the XML below so I could ingest and get my needed configuration. I have ran it through https://www.samltool.com/validate_xml.php using the Metadata XSD schema and it passes.

My SAML 2 XML Config
```xml
<EntitiesDescriptor xmlns="urn:oasis:names:tc:SAML:2.0:metadata" ID="ABC123ABC123ABC123ABC123ABC123ABC123ABC123AB">
    <ds:Signature xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
        <ds:SignedInfo>
            <ds:CanonicalizationMethod Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/>
            <ds:SignatureMethod Algorithm="http://www.w3.org/2000/09/xmldsig#rsa-sha1"/>
            <ds:Reference URI="#DEF456DEF456DEF456DEF456DEF456DEF456DEF456DE">
                <ds:Transforms>
                    <ds:Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature"/>
                    <ds:Transform Algorithm="http://www.w3.org/2001/10/xml-exc-c14n#"/>
                </ds:Transforms>
                <ds:DigestMethod Algorithm="http://www.w3.org/2000/09/xmldsig#sha1"/>
                <ds:DigestValue>xxxxxxxxxxxxxxxxxxxxxxxxxxxx</ds:DigestValue>
            </ds:Reference>
        </ds:SignedInfo>
        <ds:SignatureValue>
            xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
            xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
            xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
            xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
            xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
        </ds:SignatureValue>
        <ds:KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
            <ds:X509Data>
                <ds:X509Certificate>
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx&#13;
                    xxxxxxxxxxxx
                </ds:X509Certificate>
            </ds:X509Data>
        </ds:KeyInfo>
    </ds:Signature>
    <EntityDescriptor ID="XYZ897XYZ897XYZ897XYZ897XYZ897XYZ897XYZ8978" entityID="abcde">
        <IDPSSODescriptor ID="ASEF3854641ASEF3854641ASEF3854641ASEF3854641" WantAuthnRequestsSigned="false"
                          protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
            <KeyDescriptor use="signing">
                <ns1:KeyInfo xmlns:ns1="http://www.w3.org/2000/09/xmldsig#"
                             Id="ase74sase74sase74sase74sase74sase74sase74sse">
                    <ns1:X509Data>
                        <ns1:X509IssuerSerial>
                            <ns1:X509IssuerName>CN=test.com</ns1:X509IssuerName>
                            <ns1:X509SerialNumber>12345678978912345678</ns1:X509SerialNumber>
                        </ns1:X509IssuerSerial>
                        <ns1:X509Certificate>
                            xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
                        </ns1:X509Certificate>
                        <ns1:X509SubjectName>CN=test.com</ns1:X509SubjectName>
                    </ns1:X509Data>
                </ns1:KeyInfo>
            </KeyDescriptor>
            <SingleLogoutService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"
                                 Location="https://test.com/Logout"/>
            <SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"
                                 Location="https://test.com/affwebservices/public/saml2sso"/>
        </IDPSSODescriptor>
    </EntityDescriptor>
</EntitiesDescriptor>

```
My application error

![image](https://user-images.githubusercontent.com/446001/175677197-bf276427-adc0-4ef7-999c-f8bf0419062e.png)

My configuration is provided by passing in the XML via the `config/services`
```php
    'saml2' => [
        'metadata' => file_get_contents(storage_path('app/preprod_standard.xml')),
        'certificate' => file_get_contents(storage_path('app/preprod_standard.cer'))
    ]
```

If I make a slight alteration of the library myself in the following way, the error goes away
```php
        if ($metadata) {
            if (!Validator::make(['u' => $metadata], ['u' => 'url'])->fails()) {
                $tmp = $this->getIdentityProviderEntityDescriptorFromUrl();
            } else {
                $tmp = $this->getIdentityProviderEntityDescriptorFromXml();
            }
            if($tmp::class == \LightSaml\Model\Metadata\EntitiesDescriptor) {
                return $tmp->getAllEntityDescriptors()[0];
            }
            return $tmp;
        }
```

**Comment by 27pchrisl** [comment]
Thanks @robertdeboer ! I have submitted PR #862 to handle this.

**Comment by atymic** [comment]
Closed in #862
